docs/lib/graphqlapi/fragments/android/authz/21_oidc.md

```diff
@@ -0,0 +1,98 @@
+Add the following code to your app to initialize API plugin with OIDC auth provider:
+
+<amplify-block-switcher>
+<amplify-block name="Java">
+
+This implementation uses `CompletableFuture<T>`, which requires `minSdkVersion >= 24`.
+
+```java
+ApiAuthProviders authProviders = ApiAuthProviders.builder()
+    .oidcAuthProvider(() -> {
+        CompletableFuture<String> future = new CompletableFuture<>();
+        Amplify.Auth.fetchAuthSession(
+            session -> future.complete(((AWSCognitoAuthSession) session)
+                .getUserPoolTokens()
+                .getValue()
+                .getIdToken()),
+            future::completeExceptionally
+        );
+        try {
+            return future.get();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
```

Comment:
Do you get a compiler error, without this? Maybe you can wrap and re-throw?
```suggestion
        try {
            return future.get();
        } catch (Exception e) {
            throw new TheTypeWantedByTheProviderInterface(e);
        }
```

Reply:
Hmm, what should the recovery suggestion for the `ApiException` be in this case? Since customers are copying this code into their app, they should handle the error however they prefer to.